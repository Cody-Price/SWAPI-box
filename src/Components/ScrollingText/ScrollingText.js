import React from 'react';
import './ScrollingText.scss';

const ScrollingText = ({ crawl, title, episode }) => {

  const numeralizeEpisode = (episodeNumber) => {
    const numeralArray = ['I', 'II', 'III', 'IV', 'V', 'VI', 'VII'];
    return numeralArray[episodeNumber - 1];
  }

  if (title === undefined) {
    return ''
  } else {
    return (
      <section className="scrolling-text">
        <div className="fade"></div>
        <article className="star-wars">
          <div className="crawl">
            <div className="title">
              <p>{`Episode ${numeralizeEpisode(episode)}`}</p>
              <h1>{title}</h1>
            </div>
            <p className="crawl-text">{crawl}</p>
          </div>
        </article>
      </section>
    )
  }
}

export default ScrollingText;

// component and styling built from the CSS-Tricks article "Star Wars Crawl Text" by Geoff Graham. https://css-tricks.com/snippets/css/star-wars-crawl-text/