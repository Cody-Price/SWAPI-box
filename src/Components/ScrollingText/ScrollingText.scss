.scrolling-text {
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.title {
  text-align: center;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 60vh;
  top: -205px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  height: 400px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

@keyframes crawl {
  0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}