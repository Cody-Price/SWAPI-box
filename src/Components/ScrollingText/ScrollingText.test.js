import React from 'react';
import {shallow} from 'enzyme'; 
import ScrollingText from './ScrollingText';



describe('ScrollingText', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<ScrollingText 
      episode={5} 
      title={'A New Hope'}
      crawl={'Luke is pretty cool'}
    />)
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if nothing is passed in', () => {
    const wrapper = shallow(<ScrollingText />);
    expect(wrapper).toMatchSnapshot();
  })
})