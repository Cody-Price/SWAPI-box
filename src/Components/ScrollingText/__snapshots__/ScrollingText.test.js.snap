// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollingText
    crawl="Luke is pretty cool"
    episode={5}
    title="A New Hope"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
        />,
        <article
          className="star-wars"
        >
          <div
            className="crawl"
          >
            <div
              className="title"
            >
              <p>
                Episode V
              </p>
              <h1>
                A New Hope
              </h1>
            </div>
            <p
              className="crawl-text"
            >
              Luke is pretty cool
            </p>
          </div>
        </article>,
      ],
      "className": "scrolling-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
          >
            <div
              className="title"
            >
              <p>
                Episode V
              </p>
              <h1>
                A New Hope
              </h1>
            </div>
            <p
              className="crawl-text"
            >
              Luke is pretty cool
            </p>
          </div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                <p>
                  Episode V
                </p>
                <h1>
                  A New Hope
                </h1>
              </div>,
              <p
                className="crawl-text"
              >
                Luke is pretty cool
              </p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode V
                  </p>,
                  <h1>
                    A New Hope
                  </h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Episode V",
                  },
                  "ref": null,
                  "rendered": "Episode V",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke is pretty cool",
                "className": "crawl-text",
              },
              "ref": null,
              "rendered": "Luke is pretty cool",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
          />,
          <article
            className="star-wars"
          >
            <div
              className="crawl"
            >
              <div
                className="title"
              >
                <p>
                  Episode V
                </p>
                <h1>
                  A New Hope
                </h1>
              </div>
              <p
                className="crawl-text"
              >
                Luke is pretty cool
              </p>
            </div>
          </article>,
        ],
        "className": "scrolling-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
            >
              <div
                className="title"
              >
                <p>
                  Episode V
                </p>
                <h1>
                  A New Hope
                </h1>
              </div>
              <p
                className="crawl-text"
              >
                Luke is pretty cool
              </p>
            </div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
                >
                  <p>
                    Episode V
                  </p>
                  <h1>
                    A New Hope
                  </h1>
                </div>,
                <p
                  className="crawl-text"
                >
                  Luke is pretty cool
                </p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Episode V
                    </p>,
                    <h1>
                      A New Hope
                    </h1>,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Episode V",
                    },
                    "ref": null,
                    "rendered": "Episode V",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "A New Hope",
                    },
                    "ref": null,
                    "rendered": "A New Hope",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Luke is pretty cool",
                  "className": "crawl-text",
                },
                "ref": null,
                "rendered": "Luke is pretty cool",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ScrollingText should match the snapshot if nothing is passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollingText />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "",
  Symbol(enzyme.__nodes__): Array [
    "",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
